docs/version_upgrade_process.md

```diff
@@ -0,0 +1,70 @@
+The following guide details the steps followed in the upgrade process for IR2Vec.
```

Comment:
"The following guide details the steps followed in upgrading the LLVM version supported by IR2Vec"